src/psbt.h

```diff
@@ -424,6 +516,18 @@ struct PartiallySignedTransaction
         OverrideStream<Stream> os(&s, s.GetType(), s.GetVersion() | SERIALIZE_TRANSACTION_NO_WITNESS);
         SerializeToVector(os, *tx);
 
+        // Write xpubs
+        for (const auto xpub_pair : xpubs) {
+            for (const auto xpub : xpub_pair.second) {
```

Comment:
suggested change to reference instead of copy in range loops:
```diff
-        for (const auto xpub_pair : xpubs) {
-            for (const auto xpub : xpub_pair.second) {
+        for (const auto& xpub_pair : xpubs) {
+            for (const auto& xpub : xpub_pair.second) {
```

Reply:
(this also fixes warnings if you build with -Wrange-loop-analysis)

Reply:
Done, also in a few other places.